src/ui/categories/list/CategoryTree.js

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { Col, Spinner } from 'patternfly-react';
+import CategoryFolderIcon from 'ui/categories/common/CategoryFolderIcon';
+import CategoryExpandedIcon from 'ui/categories/common/CategoryExpandedIcon';
+import CategoryListMenuActions from 'ui/categories/list/CategoryListMenuActions';
+import RowSpinner from 'ui/pages/common/RowSpinner';
+
+class CategoryTree extends Component {
+  componentWillMount() {
+    if (this.props.onWillMount) this.props.onWillMount(this.props);
+  }
+
+  renderRows() {
+    const { categories } = this.props;
+    return categories.map((category, i) => {
+      const onClickExpand = () => {
+        if (!category.isEmpty) {
+          this.props.onExpandCategory(category.code);
+        }
+      };
+      const className = ['CategoryTree__tree-column-td'];
+      if (category.isEmpty) {
+        className.push('CategoryTree__tree-column-td--empty');
+      }
+
+      return (
+        <tr key={category.code} className="CategoryTree__row">
+          <td className={className.join(' ').trim()}>
+            <span
+              role="button"
+              tabIndex={i}
+              className="CategoryTree__icons-label"
+              style={{ marginLeft: category.depth * 24 }}
+              onClick={onClickExpand}
+              onKeyDown={onClickExpand}
+            >
+              <CategoryExpandedIcon expanded={category.expanded} />
+              <CategoryFolderIcon empty={category.isEmpty} />
+              <span className="CategoryTree__category-name">
+                { category.title }
+              </span>
+              <RowSpinner loading={!!category.loading} />
+            </span>
+          </td>
+          {/* to move into CategoryListMenuActions */}
```

Comment:
I guess this comment should be removed, because CategoryListMenuActions is into place